src/main/java/io/github/thebusybiscuit/slimefun4/api/items/SlimefunItem.java

```diff
@@ -649,6 +649,11 @@ private void checkForDeprecations(@Nullable Class<?> c) {
      *            The new {@link Research} for this {@link SlimefunItem}, or null
      */
     public void setResearch(@Nullable Research research) {
+        if (this.isDisabled()) {
```

Comment:
If this check is not here, we'd have to check every affected item when getting enabled researches.

I don't think this is a good solution anymore though tbh